hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -970,18 +967,14 @@ public static HoodieData<HoodieRecord> convertFilesToColumnStatsRecords(HoodieEn
     final List<Tuple3<String, String, Boolean>> partitionFileFlagTupleList = fetchPartitionFileInfoTriplets(partitionToDeletedFiles, partitionToAppendedFiles);
 
     // Create records MDT
+    Option<Schema> writerSchemaOpt = fetchStatsForLogFiles ? tryResolveSchemaForTable(dataMetaClient) : Option.empty();
```

Comment:
Option(write schema ) needs to be set while instantiating HoodieLogRecord. I have taken a stab to fix it. let me know .

Reply:
The log block itself contains the writer schema so it can be fetched while scanning the log blocks.  Additionally using table schema resolver is heavy and unnecessary.

Reply:
We can revisit this if not easily fixable in this PR.

Reply:
https://issues.apache.org/jira/browse/HUDI-8445

Reply:
The reason I asked about this is that the log blocks and files can be written by different commits using different writer schemas in cas there is schema evolution.  Strictly speaking, the column stats of a log file should only contain the columns that exist in the file.